pymatgen/core/structure.py

```diff
@@ -4498,6 +4504,266 @@ def substitute(self, index: int, func_group: IMolecule | Molecule | str, bond_or
         for site in func_grp[1:]:
             self._sites.append(site)
 
+    def relax(
+        self,
+        calculator: str | Calculator = "gfn2-xtb",
+        optimizer: str | Optimizer = "FIRE",
+        steps: int = 500,
+        fmax: float = 0.1,
+        opt_kwargs: dict = None,
+        return_trajectory: bool = False,
+        verbose: bool = False,
+    ) -> Molecule | tuple[Molecule, TrajectoryObserver | Trajectory]:
+        """
+        Performs a molecule relaxation using an ASE calculator.
+
+        Args:
+            calculator: An ASE Calculator or a string from the following options: "m3gnet.
+                Defaults to 'm3gnet', i.e. the M3GNet universal potential.
+            optimizer (str): name of the ASE optimizer class to use
+            steps (int): max number of steps for relaxation. Defaults to 500.
+            fmax (float): total force tolerance for relaxation convergence.
+                Defaults to 0.1 eV/A.
+            opt_kwargs (dict): kwargs for the ASE optimizer class.
+            return_trajectory (bool): Whether to return the trajectory of relaxation.
+                Defaults to False.
+            verbose (bool): whether to print out relaxation steps. Defaults to False.
+
+        Returns:
+            Molecule: Relaxed molecule
+        """
+        return _relax(
+            self,
+            calculator,
+            relax_cell=False,
+            optimizer=optimizer,
+            steps=steps,
+            fmax=fmax,
+            opt_kwargs=opt_kwargs,
+            return_trajectory=return_trajectory,
+            verbose=verbose,
+        )
+
+    def calculate(
+        self,
+        calculator: str | Calculator = "gfn2-xtb",
+        verbose: bool = False,
+    ) -> Molecule:
+        """
+        Performs an ASE calculation.
+
+        Args:
+            calculator: An ASE Calculator or a string from the following options: "m3gnet.
+                Defaults to 'm3gnet', i.e. the M3GNet universal potential.
+            verbose (bool): whether to print stdout. Defaults to False.
+
+        Returns:
+            Molecule: Molecule following ASE calculation.
+        """
+        return _calculate(self, calculator, verbose=verbose)
+
+
+def _calculate(
```

Comment:
Would it maybe make more sense to add these private functions as methods to `class SiteCollection`?

Reply:
Ah, I was looking for something that was inherited by both Structure and Molecule. Didn't know about this. I think it makes more sense to add there.